lib/job-simulation.rb

```diff
@@ -5,12 +5,52 @@ class JobSimulation
   attr_reader :workers, :waiting, :roll
 
   def initialize (jobs_available, job_seekers)
+    # assume jobs_available and job_seekers are ints
+    # assume always more job_seekers than jobs_available
+    raise ArgumentError.new("The economy is bad. There will always be more job seekers than jobs.") if jobs_available > job_seekers
 
+    @workers = Stack.new
+    (1..jobs_available).each do |worker|
+      @workers.push("Worker ##{worker}")
+    end
+
+    @waiting = Queue.new
+    # seekers_left = job_seekers - jobs_available
+    ((jobs_available + 1)..job_seekers).each do |worker|
+      @waiting.enqueue("Worker ##{worker}")
+    end
+
+    @roll
   end
 
   def cycle
+    roll
```

Comment:
Just put the rolling method body here - it's only one line, so no need for a new method for it